src/literalset.rs

```diff
@@ -40,6 +40,18 @@ impl<'l> TrimmedLiteralRef<'l> {
     pub fn as_ref(&self) -> &TrimmedLiteral {
         self.reference
     }
+    pub fn truncate(&self, min_chars: usize, max_chars: usize) -> &str {
+        let mut s = &self.reference.as_str();
+        let idy = match s.char_indices().nth(min_chars) {
+            None => 0,
+            Some((idy, _)) => idy,
+        };
+        let idx = match s.char_indices().nth(max_chars) {
+            None => s.char_indices().count() - 1,
```

Comment:
I am not sure if we should use `s.len()`

```
    println!("{}", "é язык 😋 ックス".len()); // 26
    println!("{}", "é язык 😋 ックス".graphemes(true).count()); // 12
    println!("{}", "é язык 😋 ックス".char_indices().count()); // 12
    println!("{}", "é язык 😋 ックス".chars().count()); // 12
```
Take a look at: https://crates.io/crates/unicode-segmentation
This is what is more recommended but I have not faced mismatch with `char_indices()

https://stackoverflow.com/questions/46290655/get-the-string-length-in-characters-in-rust

Reply:
You are correct. But in that case, we should consider extracting it into a `fn len_in_chars()` so in a follow up PR we can implement a caching of byte index to unicode char to simplify the lookup.

Reply:
I will pay attention next time. This time, I was trying it out first to see if it compiles or if I would have to change more in the same fn then I wanted to have in one specific commit. I will commit next time so I do not forget :)

> A few more operational things to ease the review apply process in case you are not aware yet:
> 

> * under the `Changes` "tab" you can pick my suggested changes and apply them all as one commit
> * mark conversations you resolved by clicking the resolved button

Reply:
should I respond the conversation and already resolve or wait for you to see the comments? Which do you prefer? let me know =)

Reply:
Comment + resolve is fine if there is nothing left to discuss, if you added additional thoughts in that comment, don't resolve just yet :)

Reply:
> You are correct. But in that case, we should consider extracting it into a `fn len_in_chars()` so in a follow up PR we can implement a caching of byte index to unicode char to simplify the lookup.

Don't take the above as criticism :) it's really just a performance thing (see #57 though it's a bit sparse on details just yet) - your impl is good!